intercom2/client.py

```diff
@@ -28,27 +29,20 @@ def __init__(self, token, max_retries=8, delay=5):
             "Content-Type": "application/json",
             "Intercom-Version": "2.0"
         })
-        self.MAX_RETRIES = max_retries
-        self.DELAY = delay
+        retry_strategy = Retry(
+            total=max_retries,
+            backoff_factor=delay,
+            status_forcelist=[429]
```

Comment:
for intercom POST's are retryable, so we should add:

`allowed_methods=frozenset({'DELETE', 'GET', 'HEAD', 'OPTIONS', 'PUT', 'TRACE', 'POST'})`

Otherwise we are changing the functionality for the post methods below.

Also.   Thanks, I didn't know this utility existed!